src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentManagerImpl.java

```diff
@@ -0,0 +1,389 @@
+            UUID participantEKID = Preconditions
+                    .checkNotNull( commonTasksManager
+                                    .getParticipantEntityKeyId( organizationId, studyId, participantId ),
+                            "participant not found" );
+
+            // filtered search on participants to get associated study entities
+            Map<UUID, List<NeighborEntityDetails>> neighborResults = searchApi.executeFilteredEntityNeighborSearch(
+                    participantsESID,
+                    new EntityNeighborsFilter(
+                            ImmutableSet.of( participantEKID ),
+                            java.util.Optional.of( ImmutableSet.of() ),
+                            java.util.Optional.of( ImmutableSet.of( studiesESID ) ),
+                            java.util.Optional.of( ImmutableSet.of( participatedInESID ) )
+                    )
+            );
+
+            return neighborResults.getOrDefault( participantEKID, List.of() )
+                    .stream()
+                    .filter( neighbor -> studyId.toString()
+                            .equals( getFirstValueOrNull( neighbor.getNeighborDetails().orElse( Map.of() ),
+                                    STRING_ID_FQN ) ) )
+                    .map( neighbor -> neighbor.getAssociationDetails()
+                            .getOrDefault( STATUS_FQN, Set.of( ParticipationStatus.UNKNOWN.toString() ) ).iterator()
+                            .next().toString() )
+                    .map( ParticipationStatus::valueOf )
+                    .findFirst().orElse( ParticipationStatus.UNKNOWN );
+
+        } catch ( Exception e ) {
+            logger.error( "unable to get participation status for participant: {}, study: {}, organization: {}.",
+                    participantId,
+                    studyId,
+                    organizationId, e );
+            return ParticipationStatus.UNKNOWN;
+        }
+    }
+
+    @Override
+    public boolean isNotificationsEnabled( UUID organizationId, UUID studyId ) {
+        logger.info( "Checking notifications enabled on studyId = {}, organization = {}", studyId, organizationId );
+
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // get entity set ids
+            UUID studyESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID notificationESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE, NOTIFICATION, NOTIFICATION_ES );
+            UUID partOfESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PART_OF, PART_OF_ES );
+
+            // ensure study exists
+            UUID studyEKID = Preconditions
+                    .checkNotNull( commonTasksManager.getStudyEntityKeyId( organizationId, studyId ),
+                            "study entity must exist" );
```

Comment:
should log ekid

Reply:
probably meant organizationId and studyId